docs/data/joy/components/autocomplete/autocomplete.md

```diff
@@ -0,0 +1,355 @@
+- `clearOnBlur` to help the user enter a new value.
+- `handleHomeEndKeys` to move focus inside the popup with the <kbd class="key">Home</kbd> and <kbd class="key">End</kbd> keys.
+- A last option, for instance: `Add "YOUR SEARCH"`.
+
+{{"demo": "FreeSoloCreateOption.js"}}
+
+You could also display a dialog when the user wants to add a new value.
+
+{{"demo": "FreeSoloCreateOptionDialog.js"}}
+
+## Option appearance
+
+To customize the appearance of the options, use `renderOption` prop in combination with the `AutocompleteOption` component as an option container.
+
+{{"demo": "CountrySelect.js"}}
+
+:::info
+The `AutocompleteOption` uses the same styles and variables as [`ListItemButton`](/joy-ui/react-list/#actionable), so that you get the same customization experience.
+:::
+
+## Input appearance
+
+To customize the appearance of the input, use `renderInput` prop that returns an `Input` component.
+
+{{"demo": "InputAppearance.js"}}
+
+## Validation
+
+To display invalid state, set the `error` prop on the `FormControl`.
+
+{{"demo": "AutocompleteError.js"}}
+
+## Grouped
+
+You can group the options with the `groupBy` prop.
+If you do so, make sure that the options are also sorted with the same dimension that they are grouped by,
+otherwise, you will notice duplicate headers.
+
+{{"demo": "Grouped.js"}}
+
+## Disabled options
+
+{{"demo": "DisabledOptions.js"}}
+
+## Asynchronous requests
+
+The component supports two different asynchronous use-cases:
+
+- [Load on open](#load-on-open): it waits for the component to be interacted with to load the options.
+- [Search as you type](#search-as-you-type): a new request is made for each keystroke.
+
+### Load on open
+
+It displays a progress state as long as the network request is pending.
+
+{{"demo": "Asynchronous.js"}}
+
+### Search as you type
+
+If your logic is fetching new options on each keystroke and using the current value of the textbox
+to filter on the server, you may want to consider throttling requests.
+
+Additionally, you will need to disable the built-in filtering of the `Autocomplete` component by
+overriding the `filterOptions` prop:
+
+```jsx
+<Autocomplete filterOptions={(x) => x} />
+```
+
+<!-- ### Google Maps place
+
+A customized UI for Google Maps Places Autocomplete.
+For this demo, we need to load the [Google Maps JavaScript](https://developers.google.com/maps/documentation/javascript/overview) and [Google Places](https://developers.google.com/maps/documentation/places/web-service/overview) API. -->
+
+<!-- {{"demo": "GoogleMaps.js"}} -->
+
+<!-- :::warning
+⚠️ Before you can start using the Google Maps JavaScript API and Places API, you must sign up and create a billing account.
+::: -->
+
+## Multiple values
+
+Also known as tags, the user is allowed to enter more than one value. By default, the autocomplete uses [`Chip`](/joy-ui/react-chip/) component to render tags.
+
+{{"demo": "Tags.js"}}
+
+### Tag appearance
```

Comment:
I wonder if we shouldn't take care of the terminology here. Terms like "badge", "chip", and "tag" are used out there interchangeably and given we already have a component called `Chip`, I think we should probably not add a second term to refer to it. So, I believe everything here regarding _tags_, should actually be _chip_ ⎯`renderChip` & `limitChips`.

Reply:
How about `renderSelectedOptions`? this sounds more general than "badge", "chip", or "tag" to me.

Reply:
Are you seeing any potential bad consequences of using straight-up "chips"?

Reply:
cc @samuelsycamore @oliviertassinari @mnajdova for more thoughts around this.

Reply:
I like `renderSelectedOptions`. My reasons for it, it describes what it is, not how it looks. In my custom design system, they may not look like chips :)